### PR TITLE
Reproducible Verification of Stylus Programs

### PR DESCRIPTION
**Pull request by rauljordan**
This PR carries out the work from #25 merged with the latest Stylus changes. It includes the ability to run every cargo stylus command in a Docker container by Rust version `cargo stylus reproducible 1.77 deploy --private-key=$PRIV_KEY`, and for verifying deployed Stylus programs against a local project with `cargo stylus verify --deployment-tx=$DEPLOYMENT_TX_HASH`

### CODE REVIEW

File: check/src/project.rs [hunk 1]
```diff
@@ -7,17 +7,26 @@ use crate::{
 };
```
Comment (rauljordan) [inline check/src/project.rs:7]: Review this file

File: check/src/verify.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2023, Offchain Labs, Inc.
```
Comment (rauljordan) [inline check/src/verify.rs:1]: Review this file

Comment (joshuacolvin0) [inline check/src/verify.rs:1]:
```suggestion
// Copyright 2023-2024, Offchain Labs, Inc.
```

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Comment by rauljordan** [comment]
CI agent seems to be flawed. Will merge and check if other PRs also fail and address

**Pull request merged**